internal/impl/kube.go

```diff
@@ -453,6 +466,74 @@ func GenerateKubeDeployment(image string, dep *protos.Deployment, cfg *KubeConfi
 	return nil
 }
 
+// generateRolesAndBindings generates Kubernetes roles and role bindings that
+// grant permissions to the appropriate service accounts.
+func generateRolesAndBindings() ([]byte, error) {
+	// Grant the default service account the permission to get, list, and watch
+	// pods. The babysitter watches pods to generate routing info.
+	//
+	// TODO(mwhittaker): This leaks permissions to the user's code. We should
+	// avoid that. We might have to run the babysitter and weavelet in separate
+	// containers or pods.
+	role := rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "Role",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pods-getter",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"pods"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		},
+	}
+
+	binding := rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "RoleBinding",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default-pods-getter",
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     "pods-getter",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind: "ServiceAccount",
+				Name: "default",
+			},
+		},
+	}
+
+	var b bytes.Buffer
+	fmt.Fprintln(&b, "# Roles and bindings.")
+
+	bytes, err := yaml.Marshal(role)
+	if err != nil {
+		return nil, err
+	}
+	b.Write(bytes)
+	fmt.Fprintf(&b, "\n---\n\n")
+
+	bytes, err = yaml.Marshal(binding)
+	if err != nil {
+		return nil, err
+	}
+	b.Write(bytes)
+	fmt.Fprintf(&b, "\n---\n\n")
+
+	fmt.Fprintf(os.Stderr, "Generated roles and bindings\n")
```

Comment:
delete?

Reply:
Happy to delete, but I added it because the other functions (e.g., `generateAppDeployment`, `generateJaegerDeployment`) have print statements like this.